### ISSUE TITLE
Extend `ContrastChecker` with link-color support

### ISSUE THREAD

**Issue by aristath** [WordPress/gutenberg#24595]
Came up in https://github.com/WordPress/gutenberg/pull/24513#issuecomment-674714119

Currently `ContrastChecker` only checks for contrast between background-color and text-color.
With the introduction of link-color controls, we'll need to extend that functionality to link-colors as well.
Links however can be a little more complicated than normal text... Links should have sufficient contrast with the background color - just like normal text, but at the same time, they should have a minimum of 3:1 contrast with surrounding text.

See this [article on WCAG and link-colors](https://webaim.org/blog/wcag-2-0-and-link-colors/) for more details on what the rules are for links.
